tests/sdk_test/sdk_test.cpp

```diff
@@ -2110,9 +2116,193 @@ void MegaChatApiTest::TEST_Calls(unsigned int a1, unsigned int a2)
+
+    // A call B and B hangs up the call
+    bool *flagRequest = &requestFlagsChat[a1][MegaChatRequest::TYPE_START_CHAT_CALL]; *flagRequest = false;
+    bool *callReceived = &mCallReceived[a2]; *callReceived = false;
+    mChatIdRingInCall[a2] = MEGACHAT_INVALID_HANDLE;
+    megaChatApi[a1]->startChatCall(chatid, true);
+    ASSERT_CHAT_TEST(waitForResponse(flagRequest), "Timeout after start chat call " + std::to_string(maxTimeout) + " seconds");
+    ASSERT_CHAT_TEST(!lastErrorChat[a1], "Failed to start chat call: " + std::to_string(lastErrorChat[a1]));
+
+    ASSERT_CHAT_TEST(waitForResponse(callReceived), "Timeout expired for receiving a call");
+    ASSERT_CHAT_TEST(mChatIdRingInCall[a2] == chatid, "Incorrect chat id at call receptor");
+
+    sleep(5);
+
+    bool *termLocal0 = &mTerminationLocal[a1]; *termLocal0 = false;
+    bool *termLocal1 = &mTerminationLocal[a2]; *termLocal1 = false;
+    bool *callDestroyed0 = &mCallDestroyed[a1]; *callDestroyed0 = false;
+    bool *callDestroyed1 = &mCallDestroyed[a2]; *callDestroyed1 = false;
+    int *termCode0 = &mTerminationCode[a1]; *termCode0 = mInvalidTermCode;
+    int *termCode1 = &mTerminationCode[a2]; *termCode1 = mInvalidTermCode;
+    flagRequest = &requestFlagsChat[a2][MegaChatRequest::TYPE_HANG_CHAT_CALL]; *flagRequest = false;
+    megaChatApi[a2]->hangChatCall(chatid);
+    ASSERT_CHAT_TEST(waitForResponse(flagRequest), "Timeout after hang up chat call " + std::to_string(maxTimeout) + " seconds");
+    ASSERT_CHAT_TEST(!lastErrorChat[a2], "Failed to hang up chat call: " + std::to_string(lastErrorChat[a2]));
+    ASSERT_CHAT_TEST(waitForResponse(callDestroyed0), "The call has to be finished account 1");
+    ASSERT_CHAT_TEST(waitForResponse(callDestroyed1), "The call has to be finished account 2");
+
+    ASSERT_CHAT_TEST(*termCode0 == mRejectCallTermCode && *termCode0 == *termCode1,
+                     "Invalid Termination code. TermCode1: "
+                     + std::to_string(*termCode0)
+                     + "  TermCode2: "
+                     + std::to_string(*termCode1));
+
+    ASSERT_CHAT_TEST(*termLocal0 == false && *termLocal0 != *termLocal1, "Invalid Termination local");
+
+    // A call B and A hangs up the call before B anser
+    flagRequest = &requestFlagsChat[a1][MegaChatRequest::TYPE_START_CHAT_CALL]; *flagRequest = false;
+    callReceived = &mCallReceived[a2]; *callReceived = false;
+    mChatIdRingInCall[a2] = MEGACHAT_INVALID_HANDLE;
+    megaChatApi[a1]->startChatCall(chatid, true);
+    ASSERT_CHAT_TEST(waitForResponse(flagRequest), "Timeout after start chat call " + std::to_string(maxTimeout) + " seconds");
+    ASSERT_CHAT_TEST(!lastErrorChat[a1], "Failed to start chat call: " + std::to_string(lastErrorChat[a1]));
+
+    ASSERT_CHAT_TEST(waitForResponse(callReceived), "Timeout expired for receiving a call");
+    ASSERT_CHAT_TEST(mChatIdRingInCall[a2] == chatid, "Incorrect chat id at call receptor");
```

Comment:
Same than above: check caller status changes to ringout, and callids match at both sides.